compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.cpp

```diff
@@ -92,6 +92,76 @@ SizesAndScalableFlags TilingConfig::getVectorTileSizes() {
   return std::make_pair(vectorSizes, scalableFlags);
 }
 
+IREE::Codegen::LoweringConfigAttr
+TilingConfig::withNewVectorSizes(ArrayRef<int64_t> newSizes,
+                                 ArrayRef<bool> newScalableFlags) {
+  unsigned numDims = getNumDimensions();
+  assert(newSizes.size() == numDims &&
+         "expected `sizes` to match number of dimensions");
+  assert((newScalableFlags.empty() || newScalableFlags.size() == numDims) &&
+         "expected `scalableFlags` to match "
+         "number of dimensions (or be empty)");
+
+  auto tilingLevels = loweringConfig.getTilingLevels();
+  auto tilingLevelHasSize = [tilingLevels](unsigned tilingLevelIndex,
+                                           unsigned dimPos) mutable {
+    auto level = tilingLevels[tilingLevelIndex];
+    return level.getSizes()[dimPos] != 0;
+  };
+  auto tilingLevelForDimPos = [&](unsigned dimPos) -> unsigned {
+    unsigned parellelCommonLevel = getVectorCommonParallelLevel();
+    unsigned reductionLevel = getVectorReductionLevel();
+    if (tilingLevelHasSize(parellelCommonLevel, dimPos))
+      return parellelCommonLevel;
+    if (tilingLevelHasSize(reductionLevel, dimPos))
+      return reductionLevel;
+    if (hasVectorInnerParallelLevel()) {
+      unsigned parallelInnerLevel = getVectorInnerParallelLevel();
+      if (tilingLevelHasSize(parallelInnerLevel, dimPos))
+        return parallelInnerLevel;
+    }
+    llvm_unreachable("no vector size found for `dimPos`");
+  };
+
+  // Make a map from tiling levels to vector dims at that level.
+  std::array<SmallVector<unsigned, 4>, MaxNumTileLevels> tilingLevelToDimsMap;
+  for (unsigned dimPos = 0; dimPos < numDims; ++dimPos) {
+    auto tilingLevelIndex = tilingLevelForDimPos(dimPos);
+    tilingLevelToDimsMap[tilingLevelIndex].push_back(dimPos);
+  }
+
+  MLIRContext *context = loweringConfig.getContext();
+  SmallVector<IREE::Codegen::LoweringConfigTilingLevelAttr>
+      updatedTilingLevelsList(tilingLevels.begin(), tilingLevels.end());
```

Comment:
[nit] "new" rather than "updated"? To match with the function name - this way it would be clearer that this where the "new" stuff is being created.